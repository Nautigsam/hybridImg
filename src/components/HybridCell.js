import React from 'react';
import { Icon, Form } from 'react-bulma-components';
const File = Form.InputFile;
import StoreProvider from '../store/StoreProvider';
import { createHybrid, deleteHybrid } from '../store/StoreActions';
import HybridDisplay from './HybridDisplay';

export const HybridCell = ({
  edit,
  grid,
  line,
  col,
  dispatchToStore,
  hybrid,
}) => {
  const onHybridUpload = async e => {
    const file = e.target.files[0];

    if (!hybrid) {
      await dispatchToStore(
        createHybrid({
          file,
          name: `${line.name}/${col.name}`,
          tags: [line.id, col.id],
          grid: grid,
        })
      );
    }
  };

  const imgSize = 128;

  return (
    <td>
      {hybrid ? (
        <div style={{ position: 'relative', minWidth: `${imgSize}px` }}>
          <HybridDisplay hybrid={hybrid} hideTags />
          {edit ? (
            <span
              className="delete is-small"
              style={{
                position: 'absolute',
                top: '4px',
                right: '4px',
              }}
              onClick={() => {
                dispatchToStore(deleteHybrid(hybrid.id));
              }}
            ></span>
          ) : (
            <></>
          )}
        </div>
      ) : (
        <File
          input-props={{ accept: 'image/*' }}
          label=""
          icon={<Icon className="fa fa-image" />}
          onChange={onHybridUpload}
        />
      )}
    </td>
  );
};

function extraProps(store, props) {
  return {
    hybrid: store.getGridHybrid(props),
  };
}

export default StoreProvider(extraProps)(HybridCell);
