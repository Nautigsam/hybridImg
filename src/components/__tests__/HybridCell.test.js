import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { HybridCell } from '../HybridCell';

import renderer from 'react-test-renderer';

describe('HybridCell', () => {
  it('renders correctly', () => {
    const tags = [{ id: 'T1', name: 'T1' }, { id: 'T2', name: 'T2' }];
    const props = {
      edit: false,
      grid: 'G1',
      line: tags[0],
      col: tags[1],
    };
    const props2 = { ...props, edit: true };
    const props3 = {
      ...props,
      hybrid: {
        id: 'H1',
        name: 'Spaghetti/Cataclysme',
        url: './Images/20200409_174234.jpg',
        tags: tags,
        grid: 'G1',
        user: 'U1',
      },
    };
    const props4 = { ...props3, edit: true };
    const tree = renderer
      .create(
        <StaticRouter>
          <HybridCell {...props} />
          <HybridCell {...props2} />
          <HybridCell {...props3} />
          <HybridCell {...props4} />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
