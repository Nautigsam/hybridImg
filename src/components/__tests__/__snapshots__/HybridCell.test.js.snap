// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridCell renders correctly 1`] = `
Array [
  <td>
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "128px",
          "width": "128px",
        }
      }
      tabIndex={0}
    >
      <span
        className="icon fa fa-image"
      >
        <i
          className="rbc"
        />
      </span>
    </button>
  </td>,
  <td>
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "height": "128px",
          "width": "128px",
        }
      }
      tabIndex={0}
    >
      <span
        className="icon fa fa-image"
      >
        <i
          className="rbc"
        />
      </span>
    </button>
  </td>,
  <td>
    <div
      style={
        Object {
          "minWidth": "128px",
          "position": "relative",
        }
      }
    >
      <div
        className="box"
      >
        <a
          href="/hybrid/H1"
          onClick={[Function]}
        >
          <figure
            className="image"
          >
            <img
              alt=""
              className=""
              onError={[Function]}
              src="./Images/20200409_174234.jpg"
            />
          </figure>
        </a>
        <div
          className="level"
          style={
            Object {
              "display": "none",
              "flexWrap": "wrap",
            }
          }
        >
          <span
            className="tag is-info"
          >
            T1
          </span>
          <span
            className="tag is-info"
          >
            T2
          </span>
        </div>
      </div>
    </div>
  </td>,
  <td>
    <div
      style={
        Object {
          "minWidth": "128px",
          "position": "relative",
        }
      }
    >
      <div
        className="box"
      >
        <a
          href="/hybrid/H1"
          onClick={[Function]}
        >
          <figure
            className="image"
          >
            <img
              alt=""
              className=""
              onError={[Function]}
              src="./Images/20200409_174234.jpg"
            />
          </figure>
        </a>
        <div
          className="level"
          style={
            Object {
              "display": "none",
              "flexWrap": "wrap",
            }
          }
        >
          <span
            className="tag is-info"
          >
            T1
          </span>
          <span
            className="tag is-info"
          >
            T2
          </span>
        </div>
      </div>
      <span
        className="delete is-small"
        onClick={[Function]}
        style={
          Object {
            "position": "absolute",
            "right": "4px",
            "top": "4px",
          }
        }
      />
    </div>
  </td>,
]
`;
